src/Orchard/Tasks/Locking/Services/IDistributedLockService.cs

```diff
@@ -0,0 +1,32 @@
+﻿using System;
+
+namespace Orchard.Tasks.Locking.Services {
+    /// <summary>
+    /// Provides distributed locking functionality.
+    /// </summary>
+    public interface IDistributedLockService : ISingletonDependency {
+        /// <summary>
+        /// Tries to acquire a lock on the specified name.
+        /// </summary>
+        /// <param name="name">The name to use for the lock.</param>
+        /// <param name="maxLifetime">The maximum amount of time the lock is allowed. This is a safety net in case the caller fails to release the lock.</param>
```

Comment:
How about `maxValidFor` as an alternative name? It maps better with the `ValidUntilUtc` database column name.